src/qtism/runtime/tests/AssessmentTestSession.php

```diff
@@ -2406,10 +2406,12 @@ protected function nextRouteItem($ignoreBranchings = false, $ignorePreConditions
         $stop = false;
 
         while ($route->valid() === true && $stop === false) {
+            $branchRules = $route->current()->getEffectiveBranchRules();
+            $numberOfBranchRules = $branchRules->count();
+
             // Branchings?
-            if ($ignoreBranchings === false && count($route->current()->getBranchRules()) > 0 && $this->mustApplyBranchRules() === true) {
-                $branchRules = $route->current()->getBranchRules();
-                for ($i = 0; $i < count($branchRules); $i++) {
+            if ($ignoreBranchings === false && $numberOfBranchRules > 0 && $this->mustApplyBranchRules() === true) {
```

Comment:
Same as https://github.com/oat-sa/qti-sdk/pull/367/files#r1338455110: The variables are not type hinted, but [mustApplyBranchRules() is](https://github.com/oat-sa/qti-sdk/blob/aba11a0f23b931427be486ac73e11ed52ee35105/src/qtism/runtime/tests/AssessmentTestSession.php#L3111).

```suggestion
            if ($ignoreBranchings === false && $numberOfBranchRules > 0 && $this->mustApplyBranchRules()) {
```

Reply:
I agree. I am just trying to avoid refactor this lib to make the PR easy to understand the relevant changes.